### ISSUE TITLE
WCAG Name, Role, Value - Ensure Panel Options form element has a label

### ISSUE THREAD

**Issue by JonCole** [grafana/grafana#106227]
**Steps to reproduce**:
"[Check out Accessibility Insights!](https://nam06.safelinks.protection.outlook.com/?url=https://accessibilityinsights.io/&data=02%7c01%7cv-stfe%40microsoft.com%7cb67b2c4b646d4f9561a208d6f4b5c39b%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c636965458850501301&sdata=mxhokIKNMb22llsjXHLgU3XZibj1Qfx37rpY4PU2sfE%3D&reserved=0) - Identify accessibility bugs before check-in and make bug fixing faster and easier."

1. Create a new dashboard
2. Add a panel
3. click Edit Panel to pop out the Panel Options side-bar
4. Run the Accessibility Insights for Web tool with the FastPass option
5. Notice the accessibility issues automatically identified by the tool.

**Actual Result**:
![Image](https://github.com/user-attachments/assets/5aa47c31-39af-4338-a313-da71024b2ae0)

**Expected Result**

No failures from Accessibility Insights for Web FastPass test run

**Relevant WCAG Criteria:** [#.#.# WCAG Criterion](link to https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0)

https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html

**Environment**:
- Grafana version: 12.0.2
- Data source type & version: n/a
- User OS & Browser:
    Chromium Edge: Version 137.0.3269.1 (Official build) dev (64-bit)
    Edition Windows 11 Pro
    Version 24H2
    OS build 26100.3476
- Others:
